src/GHC/Check/PackageDb.hs

```diff
@@ -45,6 +69,31 @@ data PackageVersion
 version :: PackageVersion -> Version
 version PackageVersion{ myVersion = MyVersion v} = v
 
+#if MIN_VERSION_ghc(9,0,1)
+-- | @getPackageVersion p@ returns the version of package @p@ that will be used in the Ghc session.
+getPackageVersion :: String -> Ghc (Maybe PackageVersion)
+getPackageVersion pName = runMaybeT $ do
+  dflags <- Monad.lift getSessionDynFlags
+  let pkgst   = unitState dflags
+      depends = explicitUnits pkgst
+
+  let explicit = do
+        pkgs <- traverse (MaybeT . return . lookupUnit pkgst) depends
+        MaybeT $ return $ find (\p -> unitPackageNameString p == pName ) pkgs
+
+      notExplicit = do
+        component <- MaybeT $ return $ lookupPackageName pkgst $ PackageName $ fromString pName
+        MaybeT $ return $ lookupUnitId pkgst (indefUnit component)
+
+  p <- explicit <|> notExplicit
+
+  return $ fromPackageConfig p
+
+fromPackageConfig :: UnitInfo -> PackageVersion
+fromPackageConfig p = PackageVersion (MyVersion $ unitPackageVersion p) (Just $ unitAbiHash p)
+
+#else
+
```

Comment:
This code duplication is pretty ugly and I should probably define versions of `lookupUnit`/`lookupPackage` etc. that are compatible with both versions instead. I've done that in the [`Development.IDE.GHC.Compat`](https://github.com/haskell/haskell-language-server/pull/1649/files#diff-a54f7055c79019e6a0bfbac7b8dcce0048de756ad4ea47d02a161177dd57d9ac) module when working on ghc9 compatibility for ghcide.

Reply:
That module from ghcide should almost live in a standalone package so that it can be reused elsewhere. But no need to extract it just for ghc-check, I am fine with the duplication for the time being